### PR TITLE
Add index ( target id, type ) on cp_job

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

**Comment by vritant** [comment]
only the tests with postgres database fail with this error:

> 14:47:45 Liquibase update Failed: Migration failed for change set db/changelog/20171017062314-add-target-index-on-job-table.xml::20171017062314-1::vrjain:
14:47:45      Reason: 
14:47:45           db/changelog/changelog-create.xml : Index Exists Precondition: idx_job_targetIDType : ERROR: column am.amcanorder does not exist
14:47:45   Position: 407
14:47:45 


basically, we recently upgraded our docker images to have postgresql-server-9.6 , which has this issue.
I will dig into this further next week.

**Comment by vritant** [comment]
@candlepin-pull-request-bot test this please

File: server/src/main/resources/db/changelog/20171017062314-add-target-index-on-job-table.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet id="20171017062314-1" author="vrjain">
+        <preConditions onFail="MARK_RAN">
```
Comment (awood) [inline server/src/main/resources/db/changelog/20171017062314-add-target-index-on-job-table.xml]: Why do we need to have a precondition on this?  Are we worried that customers/hosted have already applied this index?

Comment (mstead) [reply server/src/main/resources/db/changelog/20171017062314-add-target-index-on-job-table.xml]: IT has already applied this index to hosted. They had major issues and added it to address it.

Comment (vritant) [reply server/src/main/resources/db/changelog/20171017062314-add-target-index-on-job-table.xml]: @mstead apologies, I assumed the card had the appropriate name, so I went with that. checked with tdarby, the name matches with this PR.

**Pull request merged**